ambry-frontend/src/main/java/com/github/ambry/frontend/AmbrySecurityService.java

```diff
@@ -49,22 +55,27 @@
  * sets the respective headers on response.
  */
 class AmbrySecurityService implements SecurityService {
-
   static final Set<String> OPERATIONS = Collections.unmodifiableSet(
       Utils.getStaticFieldValuesAsStrings(Operations.class)
           .collect(Collectors.toCollection(() -> new TreeSet<>(String.CASE_INSENSITIVE_ORDER))));
+  private static final Logger logger = LoggerFactory.getLogger(AmbrySecurityService.class);
   private final FrontendConfig frontendConfig;
   private final FrontendMetrics frontendMetrics;
   private final UrlSigningService urlSigningService;
   private final HostLevelThrottler hostLevelThrottler;
+  private final QuotaManager quotaManager;
+  private final QuotaConfig quotaConfig;
   private boolean isOpen;
 
   AmbrySecurityService(FrontendConfig frontendConfig, FrontendMetrics frontendMetrics,
-      UrlSigningService urlSigningService, HostLevelThrottler hostLevelThrottler) {
+      UrlSigningService urlSigningService, HostLevelThrottler hostLevelThrottler, QuotaManager quotaManager,
+      QuotaConfig quotaConfig) {
```

Comment:
nit: might as well passing a null to quotaManager when quotaConfig.requestQuotaThrottlingEnabled is false. this way, we don't have to pass a quotaconfig to this object.